consumers.go

```diff
@@ -76,6 +76,12 @@ func (m *Manager) NewConsumerFromDefault(stream string, dflt api.ConsumerConfig,
 		return nil, fmt.Errorf("configuration validation failed: %s", strings.Join(errs, ", "))
 	}
 
+	// if we have a single filter subject in the array use the single filter string instead (which will then use the extended create request subject format)
```

Comment:
please add a todo here to remove this once cli and terraform is using update consumer - I dont want to change what people send on create.

so once those 2 are using update consumer and we use the update flag etc its a marginally safer thing to do but in cli especialy thats a bit of refactor so I dont want to forget about that

Reply:
Ok